### PR TITLE
6 typescript

### PR DESCRIPTION
**Pull request by Kingstar22**
https://kingstar22.github.io/frontend-course-2021/homeworks/nikolay.krasniuk_Kingstar22/6-typescript/dist/

### CODE REVIEW

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    });
+}
+gameLogic();
+
+const readout = setInterval(() => {
+    valueRandomLetter = getRandomLetter(letters)
+    fieldLetterEL.textContent = valueRandomLetter;
+}, 2000);
+
+function getRandomLetter(str:string):string {
```
Comment (DrSmile444) [inline homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts:42]: It's better to add spaces before types for readability.

```suggestion
function getRandomLetter(str: string): string {
```

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    return str[Math.floor(Math.random() * str.length)];
+}
+
+function getRandomNumber(min:number, max:number):number {
```
Comment (DrSmile444) [inline homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts:46]: Same here

```suggestion
function getRandomNumber(min: number, max: number): number {
```

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function changeCircleSize(num:number):void {
```
Comment (DrSmile444) [inline homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts:52]: Same here. Review all your code and fix such errors, please

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/style.css [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  font-weight: 900;
+  text-align: center;
+}
+
+.game__footer-loader {
+  width: 70px;
+  height: 15px;
+  border-radius: 20px;
+  background: linear-gradient(rgb(245, 172, 36) 0 0) left/0% 100% no-repeat rgb(173, 216, 230);
+  /* animation: p2 2s infinite steps(10); */
```
Comment (DrSmile444) [inline homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/style.css:72]: If you don't need commented code, it's better to remove it

**Review by DrSmile444 — CHANGES_REQUESTED** [review]
@Kingstar22 Github Pages link doesn't work(

```
404
File not found

The site configured at this address does not contain the requested file.

If this is your site, make sure that the filename case matches the URL.
For root URLs (like http://example.com/) you must provide an index.html file.

Read the full documentation for more information about using GitHub Pages.
```

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import './style.css';
+
+const fieldLetterEL:HTMLElement = document.querySelector('[data-characters]');
+const fieldCounterEL:HTMLElement = document.querySelector('[data-caunter-number]');
+const circleEL:HTMLElement = document.querySelector('[data-game-circle]');
+const fieldPlusPoints:HTMLElement = document.querySelector('[data-plus-points]');
+const fieldMinusPoints:HTMLElement = document.querySelector('[data-minus-points]');
+const letters: string = 'QWERTYUIOPASDFGHJKLZXCVBNM'
```
[the remark this answers is not in the document]
Comment (Kingstar22) [reply homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts:8]: did I understand correctly that the ending El is missing?

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+let counter:number = 100;
+fieldCounterEL.textContent = `${100}`;
+let valueRandomLetter:string;
+let keyPress:boolean = false;
+let scores: number;
```
[the remark this answers is not in the document]
Comment (Kingstar22) [reply homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts:13]: done

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
+import './style.css';
+
+const fieldLetterEL:HTMLElement = document.querySelector('[data-characters]');
+const fieldCounterEL:HTMLElement = document.querySelector('[data-caunter-number]');
+const circleEL:HTMLElement = document.querySelector('[data-game-circle]');
+const fieldPlusPoints:HTMLElement = document.querySelector('[data-plus-points]');
+const fieldMinusPoints:HTMLElement = document.querySelector('[data-minus-points]');
+const letters: string = 'QWERTYUIOPASDFGHJKLZXCVBNM'
+let counter:number = 100;
+fieldCounterEL.textContent = `${100}`;
```
[the remark this answers is not in the document]
Comment (Kingstar22) [reply homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts:10]: done

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+let valueRandomLetter:string;
+let keyPress:boolean = false;
+let scores: number;
+
+function gameLogic(): void {
+    document.addEventListener('keydown', (event) => {
+        let item = event.key.toUpperCase();
```
[the remark this answers is not in the document]
Comment (Kingstar22) [reply homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts:17]: done

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts (hunk 3 above)
Comment (Kingstar22) [reply homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts:52]: done)

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/style.css (hunk 4 above)
Comment (Kingstar22) [reply homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/style.css:72]: done

File: homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (DrSmile444) [reply homeworks/nikolay.krasniuk_Kingstar22/6-typescript/src/index.ts:8]: No, I meant that you missed a semicolon `;` at the end of the line.  Linters have broken for TS files, and it doesn't tell you to fix it.

It's not blocking, so we can merge this pull request)

**Review by DrSmile444 — APPROVED** [review]

**Review by DrSmile444 — APPROVED** [review]

**Pull request merged**